cli/pkg/kctrl/cmd/package/installed/create_or_update.go

```diff
@@ -145,6 +148,7 @@ func NewInstallCmd(o *CreateOrUpdateOptions, flagsFactory cmdcore.FlagsFactory)
 	cmd.Flags().StringVar(&o.serviceAccountName, "service-account-name", "", "Name of an existing service account used to install underlying package contents, optional")
 	cmd.Flags().StringVar(&o.valuesFile, "values-file", "", "The path to the configuration values file, optional")
 	cmd.Flags().BoolVar(&o.values, "values", true, "Add or keep values supplied to package install, optional")
+	cmd.Flags().BoolVar(&o.DryRun, "dry-run", false, "Print YAML for resources being applied to the cluster without applying them, optional")
```

Comment:
I think the `dry-run` concept is followed from kubectl, but in kubectl, there are 2 types of dry run, server or client, in `client`, the yamls are printed and in `server` the requests are submitted but resources are not persisted. So if we introduce dry-run with only one type (`client`), it might become a bit confusing 🤔 
I think we used `--diff-changes-yaml` in kapp due to the same reason instead of using dry-run, should we do something similar over here as well?

Reply:
maybe, `dry-run=client`??

Reply:
I do not see us using `--dry-run=server` anytime soon 🤔 
I did copy over the flag name as suggested in the issue. How do y'all feel about `--resource-yamls`?

Reply:
The help section does call it out in the description, would that not suffice?
I am hesitant to use diff run here as we have considered building an actual `diff-run` multiple times and might still do so in the future 🤔

Reply:
> The help section does call it out in the description, would that not suffice?

I mean eventually users would know that it doesn't create resources when they run it, but it sounds a bit weird.

> I am hesitant to use diff run here as we have considered building an actual diff-run multiple times and might still do so in the future

This flag would make more sense with something like a diff-run, or maybe we should just go ahead with the original and call it dry-run because that's what it is doing if we only print the resource yamls and not create the resources 😅

Reply:
This would affect how well the output can be piped I think ^
But it is already called out in the command help section and I believe folks expect resources not to be persisted when they are using the flag

Reply:
@praveenrewar This can totally evolve over time. I do not see users using this as a part of their automated workflows right away. I think this is something we can think about and be a bit more lenient when it comes to backwards compatibility while we iterate over it.
One step at a time!